File: go/4kyu/explosive-sum/main.go
package kata

func ExpSum(n uint64) uint64 {
    dp := make([]uint64, n+1)
    dp[0] = 1

    for num := uint64(1); num <= n; num++ {
        for i := num; i <= n; i++ {
            dp[i] += dp[i-num]
        }
    }

    return dp[n]
}

File: rust/4kyu/explosive-sum/main.rs
fn exp_sum(n: u64) -> u64 {
    let mut dp = vec![0; (n+1) as usize];
    dp[0] = 1;

    for num in 1..=n {
        for i in num..=n {
            dp[i as usize] += dp[(i-num) as usize];
        }
    }

    return dp[n as usize];
}